pkg/provider/notebook_resource.go

```diff
@@ -0,0 +1,248 @@
+package provider
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &NotebookResource{}
+var _ resource.ResourceWithImportState = &NotebookResource{}
+
+func NewNotebookResource() resource.Resource {
+	return &NotebookResource{}
+}
+
+// NotebookResource defines the resource implementation.
+type NotebookResource struct {
+	provider *Provider
+}
+
+func (r *NotebookResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_notebook"
+}
+
+func (r *NotebookResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Notebook resource for importing and managing Jupyter notebooks in DataRobot",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "The ID of the Notebook.",
+			},
+			"name": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "The name of the Notebook.",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"file_path": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The path to the .ipynb file to import.",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"file_hash": schema.StringAttribute{
```

Comment:
What are we going to do with file hash? 

The hash is not going to change after you import the notebook file and execute it (the execution would change the notebook file content hence the hash will be different). 

Is there a possibility that someone could expect this hash to reflect the actual state of the notebook after importing?